spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkScan.java

```diff
@@ -140,7 +140,7 @@ protected Statistics estimateStatistics(Snapshot snapshot) {
     // estimate stats using snapshot summary only for partitioned tables
     // (metadata tables are unpartitioned)
     if (!table.spec().isUnpartitioned() && filterExpressions.isEmpty()) {
-      LOG.debug("Using table metadata to estimate table statistics");
+      LOG.debug("Using snapshot metadata to estimate statistics for table {}", table.name());
```

Comment:
Should we also log snapshot id?

Reply:
Agreed, it'll help to log the snapshot here. With all the new time travel options being added (like branching and tagging), in case of any issues it'll be useful to be able to debug just by looking at the logs.

Reply:
Makes sense, added.